lib/matplotlib/backend_bases.py

```diff
@@ -3640,7 +3683,10 @@ def trigger_tool(self, name):
             Name(id) of the tool triggered from within the toolbar
 
         """
-        self.navigation.tool_trigger_event(name, sender=self)
+        try:
+            self.navigation.tool_trigger_event(name, sender=self)
+        except:
+            self.toggle_toolitem(name)
```

Comment:
Okay, sequence of events:
1. Tool A starts of toggled.
2. User clicks Tool A on the toolbar, the toolbar button has this method as its callback.
3. It calls ``tool_trigger_event`` to (un)toggle the tool.
4. If ``tool_trigger_event`` fails to (un)toggle the tool for whatever reason, i.e. because of the radio constraint not allowing us to untoggle this tool, we need to make sure the GUI reflects this.  Remember the process works differently between using the mouse to (un)toggle a tool using the GUI, whereby the GUI tells the Tool that it needs to update; and using the keyboard shortcut whereby the Tool tells the GUI it needs to update.

That said, I think I need to make sure that the exception doesn't come from within the tool's trigger method, do you have any suggestion of what type of ``Exception`` I should throw?  I did think about simply returning ``True`` or ``False``, but I think exceptions represent the more pythonic way (so far as I have been told) of doing this, I think because you don't have to worry about passing status values from deep within code.